compiler/resolution/lowerForalls.cpp

```diff
@@ -1279,4 +1421,6 @@ void lowerForallStmtsInline()
   USR_STOP();
 
   removeDeadIters();
+
+  INT_ASSERT(gRetByRefToken->firstSymExpr() == NULL);
```

Comment:
I know what this does but it's worth a comment like "Ensure RetByRefToken is no longer used"